browser/views.py

```diff
@@ -624,3 +624,71 @@ def attendee_create(request, course_id):
         'form': form,
     }
     return render(request, template, context)
+
+
+@staff_member_required
+def add_investigator_record(request, person_id):
+    person = get_object_or_404(Person, pk=person_id)
+    template = "browser/investigator_create.html"
+    form = InvestigationForm()
+
+    if request.method == 'POST':
+        if 'SaveButton_Clicked' in request.POST:
+            form = InvestigationForm(request.POST, instance=person)
+            if form.is_valid():
+                investigator = Investigator(subject=form.cleaned_data.get('subject'),
+                                            role=form.cleaned_data.get('role'),
+                                            person_id=person_id, year=form.cleaned_data.get('year'),
+                                            changed_by=request.user)
+                investigator.save()
+                return HttpResponseRedirect(reverse('person', args=(person.id,)))
+
+        if 'CancelButton_Clicked' in request.POST:
```

Comment:
cancel shouldn't be a post request, just have it link back to the person's page.

Reply:
Used JQuery to link cancel button to previous page. Removed cancel from post request.